oak_functions_service/src/lib.rs

```diff
@@ -103,7 +104,12 @@ impl OakFunctions for OakFunctionsService {
                             format!("couldn't get attestation evidence: {:?}", err),
                         )
                     })?;
-                self.instance = Some(instance);
+                if self.instance.set(instance).is_err() {
```

Comment:
Is my understanding correct that it is possible that there is a race of two requests up to this point, but at this point only one set will succeed? And that's why we need also the other check further up? Is it similar to a double checked lock in a way?

Reply:
Yeah, exactly. The check above is optimistic, but we may indeed create multiple instances. The `set()` here guarantees that only one will win and others will be thrown away.

An optimization we can do is add a version of `set` to OnceCell that calls a closure while holding a lock to create the instance. That would mean we wouldn't even create multiple instances.

But I think that can be tackled separately.